lib/rust/ensogl/core/src/display/shape/primitive/style_watch.rs

```diff
@@ -108,6 +108,18 @@ impl StyleWatchFrp {
         sampler
     }
 
+    /// Queries the style sheet for a text. Returns empty string if not found.
+    pub fn get_text<T: Into<Path>>(&self, path: T) -> frp::Sampler<String> {
+        let network = &self.network;
+        let (source, current) = self.get_internal(path);
+        frp::extend! { network
+            value   <- source.map(|t| t.text().unwrap_or_default());
+            sampler <- value.sampler();
+        }
+        source.emit(current);
+        sampler
+    }
```

Comment:
This is potentially dangereous. Returning `0` in case int was not found is sometimes OK - the color will be wrong visually. What would happen if your function returns an empty string instead of a font name? Will some default font be loaded?

Reply:
A default font will be loaded, this is handled by `typeface::font::Registry`:

https://github.com/enso-org/enso/blob/ce78f9825df1ff8669f3a0fd7eb33df564a8fc59/lib/rust/ensogl/component/text/src/typeface/font.rs#L300-L305

A screenshot of the Node Searcher if the `widget.list_view.text.font` style is removed from the stylesheet:

<img width="542" alt="Screenshot 2022-05-04 at 10 08 18" src="https://user-images.githubusercontent.com/273837/166644059-8277cf0a-8f0d-47b0-aa85-f745b0eeda93.png">